nncf/common/pruning/utils.py

```diff
@@ -636,4 +638,43 @@ def identity_mask_propagation(node: NNCFNode, graph: NNCFGraph) -> None:
         # In case for disconnected NNCFGraph
         input_masks = [None]
     assert len(input_masks) == 1
-    node.data['output_mask'] = input_masks[0]
+    input_mask = input_masks[0]
+
+    if isinstance(input_mask, dict):
+        input_mask = find_input_mask_for_node(input_mask, node)
+    node.data['output_mask'] = input_mask
```

Comment:
This will be redundant if `get_input_masks` will be implemented like here: https://github.com/openvinotoolkit/nncf/pull/1244#discussion_r948793387

Reply:
Fix.